crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -5264,12 +5264,6 @@ impl<'db> TypeInferenceBuilder<'db> {
             };
 
             match (op, result) {
-                (ast::CmpOp::Eq, Some(Type::BooleanLiteral(true))) => {
-                    return Ok(Type::BooleanLiteral(false));
-                }
-                (ast::CmpOp::NotEq, Some(Type::BooleanLiteral(false))) => {
-                    return Ok(Type::BooleanLiteral(true));
-                }
```

Comment:
Yeah, I think this removal is correct. I guess to address the TODO comments added above (where we intersect with `LiteralString`, we would need to introduce some Type method which would be true for `LiteralString`, which would mean "inhabitants of this type can never be equal to any object that doesn't inhabit this type, and all proper subtypes are single-valued." And then here, we could apply these removed cases only if all positive members of the intersection are such a type.

But I think maybe `LiteralString` is the only such type we need to care about, so a `Type` method might be overkill, we could also just special-case `LiteralString` here for now.

Reply:
> But I think maybe `LiteralString` is the only such type we need to care about, so a `Type` method might be overkill, we could also just special-case `LiteralString` here for now.

No, I don't think so -- we also want to be able to narrow `bool` to `Literal[True]` or `Literal[False]` using equality, and in due course we'll want to be able to narrow `<instance of enum class>` to `<enum member>` using equality as well (assuming the enum class doesn't have a custom `__eq__` method, in which case such narrowing would not be sound).